core/spec/mailers/shipment_mailer_spec.rb

```diff
@@ -90,4 +90,18 @@
       end
     end
   end
+
+  context 'emails contain only urls of the store where the order was made' do
+    before do
+      default_host = ActionMailer::Base.default_url_options[:host]
+      allow(Spree::Core::Engine).to receive(:frontend_available?).and_return(true)
+      allow_any_instance_of(ActionDispatch::Routing::RoutesProxy).to receive(:product_url).and_return("http://#{default_host}/products/product-slug")
+      allow_any_instance_of(ActionDispatch::Routing::RoutesProxy).to receive(:root_url).and_return("http://#{default_host}")
+    end
+
+    it 'shows proper host url in email content' do
+      shipped_email = described_class.shipped_email(shipment)
+      expect(shipped_email[:store_url].value).to eq(shipment.order.store.url)
```

Comment:
It would be better to test default url options set for the email instead of header value passed. Please have a look on this example: https://github.com/spree/spree/blob/4cc2c2967352d3fe7652bd873e492c074ffcc3da/core/spec/mailers/test_mailer_spec.rb#L26